src/modules/authentication/AuthenticationWrapper.ts

```diff
@@ -155,11 +186,48 @@ export class AuthenticationWrapper implements IAuthenticationWrapper {
       const result = await msalApplication.loginPopup(popUpRequest);
       this.storeHomeAccountId(result.account!);
       return result;
-    } catch (error) {
-      throw error;
+    }catch (error) {
+        const { errorCode } = error;
+        if(this.checkIfAuthResultError(errorCode) && !this.consentingToNewScopes){
+        this.clearCache();
+        this.deleteHomeAccountId();
+        window.sessionStorage.clear();
+        this.eraseInteractionInProgressCookie();
+        const result = await msalApplication.loginPopup(popUpRequest);
+        this.storeHomeAccountId(result.account!);
+        return result;
+      }
```

Comment:
**nitpick:** Check the indentation in this section

**suggestion:** There is a chance for a situation where the user will be in a popup loop since the error doesn't self resolve if the app tries to re-request a login popup. It could be safer to throw the error after clearing the cache since the user will get a hint about what went wrong

Reply:
That's a good idea. I missed that. I'll throw the error after clearing the cache.